gobblin-runtime/src/main/java/org/apache/gobblin/runtime/fork/Fork.java

```diff
@@ -227,7 +227,10 @@ public void consumeRecordStream(RecordStreamWithMetadata<D, S> stream)
 
           r.ack();
         }
-      }, e -> logger.error("Failed to process record.", e),
+      }, e -> {
+          logger.error("Failed to process record.", e);
+          throw(new RuntimeException(e));
```

Comment:
Any reason why it is being wrapped into RuntimeException? Can we just rethrow e as is?